packages/react-input/src/components/Input/useInputStyles.ts

```diff
@@ -1,28 +1,225 @@
+    flexGrow: 1,
+    display: 'flex',
+    alignItems: 'center',
+    gap: horizontalSpacing.xxs,
+    padding: `0 ${horizontalSpacing.mNudge}`,
+
+    // Set this on inputWrapper rather than root since bookends will have separate background colors and borders
+    background: backgroundColors.filledLighter(theme), // for outline/filledLighter
+
+    // This may need to be conditionally applied to start/end corners when bookend styling is added
+    borderRadius: borderRadius(theme),
+  }),
+  inputWrapperSmall: {
+    padding: `0 ${horizontalSpacing.sNudge}`,
+  },
+  inputWrapperLarge: {
+    padding: `0 ${horizontalSpacing.m}`,
+  },
+  inputWrapperFilled: theme => ({
+    // This is set on inputWrapper since bookends have different borders.
+    // Don't set it in the main inputWrapper style to avoid override issues for underline appearance.
+    border: `1px solid ${theme.alias.color.neutral.transparentStroke}`,
+  }),
+  inputWrapperFilledDarker: theme => ({
+    backgroundColor: backgroundColors.filledDarker(theme),
+  }),
+  inputWrapperUnderline: theme => ({
+    backgroundColor: backgroundColors.transparent(theme),
+    border: 'none', // set later
+    borderRadius: 0, // corners look strange if rounded
+  }),
+  inputWrapperOutline: theme => ({
+    border: `1px solid ${theme.alias.color.neutral.neutralStroke1}`,
+  }),
+  /** bottom border for underline/outline appearances */
+  inputWrapperBorderBottom: theme => ({
+    borderBottom: `1px solid ${theme.alias.color.neutral.neutralStrokeAccessible}`,
+  }),
+  inputWrapperDisabled: theme => ({
+    border: `1px solid ${theme.alias.color.neutral.neutralStrokeDisabled}`,
+    borderRadius: borderRadius(theme), // because underline doesn't usually have a radius
+  }),
+  inside: theme => ({
+    color: theme.alias.color.neutral.neutralForeground3, // "icon color" in design spec
+    // special case styling for icons (most common case) to ensure they're centered vertically
+    '> span > svg': { display: 'block' },
+  }),
+  insideDisabled: theme => ({
+    color: theme.alias.color.neutral.neutralForegroundDisabled,
+  }),
 });
 
 /**
  * Apply styling to the Input slots based on the state
  */
 export const useInputStyles = (state: InputState): InputState => {
+  const { size, appearance = 'outline' } = state;
+  const small = size === 'small';
+  const large = size === 'large';
+  const underline = appearance === 'underline';
+  const outline = appearance === 'outline';
+  const filledDarker = appearance === 'filledDarker';
+  const filledLighter = appearance === 'filledLighter';
+  const disabled = state.input.disabled;
+
   const styles = useStyles();
-  state.root.className = mergeClasses(styles.root, state.root.className);
 
-  // TODO Add class names to slots, for example:
-  // state.mySlot.className = mergeClasses(styles.mySlot, state.mySlot.className);
+  state.root.className = mergeClasses(
+    styles.root,
+    small && styles.rootSmall,
+    large && styles.rootLarge,
```

Comment:
One thing that has been done in other components and could be done here is split the `useStyles` call into multiple calls, i.e. `useRootStyles`, `useInputStyles`, etc. Then this line could be like this instead:

```suggestion
    rootStyles[size],
```

And it would apply the `small`, `medium` or `large` styles appropriately depending on the value of `size`. You would have to have some styles definition for `medium` though.